### PR TITLE
Add GH workflow for checking external links

### LINKED ISSUE #60: Run a periodic link checker?
JupyterBook has a tool for checking the health of all the links in the web links in the source material:
```
jupyter-book build . --builder linkcheck 
```

I wonder if we can set up a GitHub action to run this check periodically in order to flag any upstream changes so our links stay up to date.

### PR DESCRIPTION
**Pull request by andersy005**
Closes #60

### CODE REVIEW

**Comment by brian-rose** [comment]
Cool! Thanks @andersy005 

Is there a command-line flag to disable the re-execution of notebooks during the `jupyter-book build`? Seems like an unnecessary step for this test that will eat up actions hours for no good reason.

**Comment by andersy005** [comment]
> Is there a command-line flag to disable the re-execution of notebooks during the jupyter-book build?

Are you referring to this configuration option, by any chance?


https://github.com/ProjectPythia/pythia-foundations/blob/b0226d5e501b71057897e6d78daa4bf4057bfb9f/_config.yml#L10-L13

**Comment by andersy005** [comment]
> I think as the book content grows, this step is bound to get more and more expensive.

I agree. As far as I can tell, the only option for overriding the settings is to provide a path to a custom configuration file


```bash
Options:
  --path-output TEXT              Path to the output artifacts
  --config TEXT                   Path to the YAML configuration file
                                  (default: PATH_SOURCE/_config.yml)
```

Do we want to keep two versions of this file?

**Comment by brian-rose** [comment]
> Do we want to keep two versions of this file?

I guess that's a viable workaround, but could lead to issues in the future if we make other changes to `_config.yml`.

I think it would also be worth pursuing this issue upstream, so that we can do something like
```
jupyter-book build . --no-execution
``` 

I'll try to open an issue over at https://github.com/executablebooks/jupyter-book

**Comment by andersy005** [comment]
> I think it would also be worth pursuing this issue upstream, so that we can do something like

	jupyter-book build . --no-execution

> I'll try to open an issue over at executablebooks/jupyter-book

👍🏽. Let's revisit this once we've heard from the `executablebooks`' folks.

**Comment by brian-rose** [comment]
Over at https://github.com/executablebooks/jupyter-book/issues/1343, @choldgraf suggested simply updating the `config.yml` file to set `execute_notebooks: off` and writing it out to disk before running the `jupyter-book build` command.

@andersy005 could we do this as a one-liner as part of the GitHub action?

File: .github/workflows/link-checker.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+      - name: Disable notebook execution
+        shell: python
+        run: |
+          import yaml
+          with open('_config.yml') as f:
+            data = yaml.safe_load(f)
+          data['execute']['execute_notebooks'] = 'off'
+          with open('_config.yml', 'w') as f:
+            yaml.dump(data, f)
```
Comment (andersy005) [inline .github/workflows/link-checker.yaml:41]: > @andersy005 could we do this as a one-liner as part of the GitHub action?

Here's what I came up with

**Review by brian-rose — APPROVED** [review]
Nice! Glad we have an easy way to disable the execution.

File: .github/workflows/link-checker.yaml (hunk 1 above)
Comment (brian-rose) [reply .github/workflows/link-checker.yaml:41]: Seems to work as advertised!

**Pull request merged**